src/otx/core/data/dataset/segmentation.py

```diff
@@ -171,6 +171,13 @@ def __init__(
             stack_images,
             to_tv_image,
         )
+        is_polygon = isinstance(dm_subset[0].annotations[-1], (Polygon, Ellipse))
+        if is_polygon:
+            # insert background class at index 0
+            self.label_info.label_names.insert(0, "background")
+            # in semantic segmentation we have only one label_group
+            self.label_info.label_groups[-1].insert(0, "background")
```

Comment:
You are judging a dataset-level characteristic `self.label_info` by probing just one sample `dm_subset[0]` in the dataset. This logic is really uncomfortable to see and I think that we should find another way.

Reply:
I think that there has been such a problem on labeling in semantic segmentation task and we should fix the problem in a correct way. Before we do this, please answer me this first.

What is the model side restriction on the number of labels? Is it possible that the model head can handle `num_classes=1` (binary classification) case? If not, is it true that we should keep `num_classes>1` because of this restriction?

Reply:
In the current code model can't handle num_classes =1 . We should change head construction to support it.
I judge about dataset only by 1 example, because we don't have cases when we have mixed annotations. If we have polygons then we have Geti case, where we don't have background label. 
Let me think how to check it in a more appropriate way

Reply:
I added check as property relying on subset.get_annotated_type(). I think now it is better